pandas/core/indexes/category.py

```diff
@@ -429,10 +437,11 @@ def get_loc(self, key, method=None):
         >>> non_monotonic_index.get_loc('b')
         array([False,  True, False,  True], dtype=bool)
         """
-        codes = self.categories.get_loc(key)
-        if (codes == -1):
-            raise KeyError(key)
```

Comment:
codes can never be -1. If key is not in self.categories, a KeyError was always raised.